platform/core/Makefile

```diff
@@ -1,30 +1,75 @@
 .DEFAULT_GOAL := all
 
+build_type := $(if $(filter $(MODE), release),release,debug)
+release_flag := $(if $(filter $(build_type), release), --release)
```

Comment:
Including the `if`s here is a bit confusing; I had to come back here to check that effectively that was the case. *But* this is mainly a naming nit: a flag is a flag, this looks more like a *condition* or a *check*. Same above.